### DISCUSSION TITLE
如何按需引入依赖

### DISCUSSION THREAD

**Discussion by keosu** [xmake-io/xmake-vscode#223] [Q&A]
问题0： 想问一下 lua中这里的缩进是有语义的(如python)还是只是为了好看？
问题1：假设我有两个target，每个依赖不同的第三方库，add_requires好像只能放在最前面，下面这么，就会报错

add_requires("aa")
target( a)
        set_kind("binary")
        add_files("src/a/*.cpp") 
        add_packages("aa")
     

add_requires("bb")   -- 这里会报错， 只能放到最前面
target( b)
        set_kind("binary")
        add_files("src/b/*.cpp") 
        add_package("bb")
我是想按照target 按需引入 依赖，不知道一般推荐的写法是啥？谢谢

**Comment by waruqi** [comment]
这个跟 xmake-vscode 没关系，提到 xmake 里面去

**Reply by keosu** [reply]
也许你可以简要先写两句再说不合适

**Reply by waruqi** [reply]
我是说 这里是 xmake-vscode 插件工程，你这个跟插件没任何关系，不要在这里问，xmake 的问题提到 xmake 的 issues/discussion  

https://github.com/xmake-io/xmake/issues
https://github.com/xmake-io/xmake/discussions

啥合适不合适？
